src/core/components/dht.js

```diff
@@ -0,0 +1,109 @@
+'use strict'
+
+const promisify = require('promisify-es6')
+const every = require('async/every')
+const each = require('async/each')
+const bsplit = require('buffer-split')
+
+module.exports = (self) => {
+  return {
+    /**
+     * Find the closest peers to a given `PeerId`, by querying the DHT.
+     *
+     * @param {PeerId} peer - The `PeerId` to run the query agains.
+     * @param {function(Error, Array<PeerId>)} [callback]
+     * @returns {Promise<Array<PeerId>>|void}
+     */
+    query: promisify((peer, callback) => {
+      self.libp2p.dht.getClosestPeers(peer.id, callback)
+    }),
+    /**
+     * Find peers in the DHT that can provide a specific value, given a key.
+     *
+     * @param {CID} key - They key to find providers for.
+     * @param {function(Error, Array<PeerInfo>)} [callback]
+     * @returns {Promise<PeerInfo>|void}
+     */
+    findprovs: promisify((key, callback) => {
```

Comment:
nvm, let's keep using go-ipfs API for now - https://github.com/ipfs/interface-ipfs-core/issues/136